src/config.h

```diff
@@ -5,3 +5,23 @@
 #else
 	#warning Missing configuration
 #endif
+
+/*
+ * This is here because CMake won't let us a define a function-style macro from CMakeLists.
+ * And we're using a function-style macro so that using it undefined raises an error,
+ * because '#if FOO' or '#if FOO == 1' is silently false (mandated by C standard).
+ *
+ * in short:
+ * - IS_CLIENT_BUILD is for optional helpers in header files and similar
+ * - CHECK_CLIENT_BUILD() is for cases where program behavior differs
+ * - Do not use MT_BUILDTARGET directly
+ */
+#if MT_BUILDTARGET == 1
+#define IS_CLIENT_BUILD 1
+#define CHECK_CLIENT_BUILD() 1
+#elif MT_BUILDTARGET == 2
+#define IS_CLIENT_BUILD 0
+#define CHECK_CLIENT_BUILD() 0
+#else
+#define IS_CLIENT_BUILD 0
```

Comment:
shouldn't this be an error?

Reply:
no, this applies to files that are compiled once and included in both client & server binaries.